lib/services/ios-project-service.ts

```diff
@@ -300,9 +304,62 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
 
 			let xcodeBuildVersion = this.getXcodeVersion();
 			if (helpers.versionCompare(xcodeBuildVersion, "8.0") >= 0) {
-				let teamId = this.getDevelopmentTeam();
-				if (teamId) {
-					args = args.concat("DEVELOPMENT_TEAM=" + teamId);
+				// TRICKY: I am not sure why we totally disregard the buildConfig parameter here.
+				buildConfig = buildConfig || {};
+
+				if (this.$options.teamId) {
+					buildConfig.teamIdentifier = this.$options.teamId;
+				} else {
+					buildConfig = this.readXCConfigSigning();
+					if (!buildConfig.codeSignIdentity && !buildConfig.mobileProvisionIdentifier && !buildConfig.teamIdentifier) {
+						buildConfig = this.readBuildConfigFromPlatforms();
+					}
+				}
+
+				let signingStyle: XcodeSigningStyle;
+				if (buildConfig.codeSignIdentity || buildConfig.mobileProvisionIdentifier) {
+					signingStyle = "Manual";
+				} else if (buildConfig.teamIdentifier) {
+					signingStyle = "Automatic";
+				} else {
+					let signingStyles = [
+						"Manual - Select existing provisioning profile for use",
+						"Automatic - Select Team ID for signing and let Xcode select managed provisioning profile"
+					];
+					let signingStyleIndex = signingStyles.indexOf(this.$prompter.promptForChoice("Select codesiging style", signingStyles).wait());
+					signingStyle = new Array<XcodeSigningStyle>("Manual", "Automatic")[signingStyleIndex];
+
+					switch(signingStyle) {
+						case "Manual":
+							let profile = this.getProvisioningProfile();
+							if (!profile) {
+								this.$logger.error("No matching provisioning profile found.");
+							}
+							this.persistProvisioningProfiles(profile.UUID);
+							this.$logger.info("Apply provisioning profile: " + profile.Name + " (" + profile.TeamName + ") " + profile.Type + " UUID: " + profile.UUID);
+							buildConfig.mobileProvisionIdentifier = profile.UUID;
+							buildConfig.teamIdentifier = profile.TeamIdentifier[0];
+							break;
+						case "Automatic":
+							buildConfig.teamIdentifier = this.getDevelopmentTeam();
+							this.persistDevelopmentTeam(buildConfig.teamIdentifier);
+							break;
+					}
+				}
+
+				switch(signingStyle) {
+					case "Manual": {
```

Comment:
do you really need the curly brackets `{` - they seems inconsistent with the other code.